### ISSUE TITLE
Nullable issue in CreditCardPaymentResponse

### ISSUE THREAD

**Issue by martijnmelchers** [Viincenttt/MollieApi#373]
When getting a payment from the Mollie API we use the following code to communicate Creditcard Failure reason to the user: 

```csharp
if (payment is CreditCardPaymentResponse response && !string.IsNullOrWhiteSpace(response.Details.FailureMessage))
            matchingTransaction.FailureReason = response.Details.FailureMessage;
```

However when a payment is in the state Expired `Details` is null thus breaking this implementation. Should this be marked as nullable or should we guard against this better?

**Comment by Viincenttt** [comment]
Thanks for the headsup! Do you happen to know what the status of the payment is when it returns `null` for the `Details` property? I'll try to reproduce this on my end.

**Comment by martijnmelchers** [comment]
It is null when the creditcard payment is in the expired state

**Comment by Viincenttt** [comment]
Hi @martijnmelchers 

I've done some testing and indeed found that the `Details` property for all payment responses can be null, so I've marked these properties as nullable in PR #377 

Kind regards,
Vincent

**Comment by Viincenttt** [comment]
Hi @martijnmelchers 

I just released version 4.2.0.0 of the library where I've made the `Details` property nullable.

Thanks for reporting this issue!

Happy coding!
Vincent
